src/lib.rs

```diff
@@ -0,0 +1,1174 @@
+                    if tz < bits_per_block::<T>() {
+                        let bs = b * bits_per_block::<T>() + i_off + tz;
+                        self.range.start = bs + 1;
+                        return Some(bs);
+                    }
+                }
+                i = b * bits_per_block::<T>();
+            }
+        }
+        self.range.start = self.range.end;
+        None
+    }
+
+    fn size_hint(&self) -> (usize, Option<usize>) {
+        self.range.size_hint()
+    }
+}
+
+#[derive(Clone)]
+pub struct IterUnsetBits<'a, T: 'a> {
+    vob: &'a Vob<T>,
+    range: Range<usize>,
+}
+
+impl<'a, T: Debug + One + PrimInt + Zero> Iterator for IterUnsetBits<'a, T> {
+    type Item = usize;
+
+    fn next(&mut self) -> Option<usize> {
+        if let Some(mut i) = self.range.next() {
+            for b in storage_block_off::<T>(i) .. storage_blocks_required::<T>(self.vob.len) {
+                let v = self.vob.vec[b];
+                if v != T::max_value() {
+                    let i_off = i % bits_per_block::<T>();
+                    let tz = (!v >> i_off).trailing_zeros() as usize;
+                    if tz < bits_per_block::<T>() {
+                        let bs = b * bits_per_block::<T>() + i_off + tz;
+                        self.range.start = bs + 1;
+                        if bs >= self.vob.len {
+                            break;
+                        }
+                        return Some(bs);
+                    }
+                }
+                i = b * bits_per_block::<T>();
+            }
+        }
+        self.range.start = self.range.end;
+        None
+    }
+
+    fn size_hint(&self) -> (usize, Option<usize>) {
+        self.range.size_hint()
+    }
+}
+
+impl<T: Debug + One + PrimInt + Zero> PartialEq for Vob<T> {
+    fn eq(&self, other: &Self) -> bool {
+        if self.len != other.len {
+            return false;
+        }
+        self.iter_storage()
+            .zip(other.iter_storage())
+            .all(|(w1, w2)| w1 == w2)
+    }
+}
+
+impl<T: Debug + One + PrimInt + Zero> Eq for Vob<T> {}
+
+impl<T :Debug + Hash + One + PrimInt + Zero> Hash for Vob<T> {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        self.len.hash(state);
+        for b in self.iter_storage() {
+            b.hash(state);
+        }
+    }
+}
+
+#[derive(Clone)]
+pub struct StorageIter<'a, B: 'a> {
+    iter: slice::Iter<'a, B>,
+}
+
+impl<'a, T: Debug + One + PrimInt + Zero> Iterator for StorageIter<'a, T> {
+    type Item = T;
+
+    fn next(&mut self) -> Option<T> {
+        self.iter.next().cloned()
+    }
+
+    fn size_hint(&self) -> (usize, Option<usize>) {
+        self.iter.size_hint()
+    }
+}
+
+/// How many bits are stored in each underlying storage block?
+fn bits_per_block<T>() -> usize {
+    size_of::<T>() * 8
+}
+
+/// Return the offset in the vector of the storage block storing the bit `off`.
+fn storage_block_off<T>(off: usize) -> usize {
```

Comment:
I feel like I'm being awkward here, but the name of this one is confusing too. I thought "block off" was intended as a verb, as in "to block off the storage" (like one "blocks off a road" for example), but it's actually meant to mean "offset".

Reply:
OK, I could call them `block_offset` and `blocks_required`? [I think `off` is a very common abbreviation for "offset", but maybe I'm wrong.]

Reply:
It is yes, it's just an unfortunate interaction with "block off".

`block_offset` could work, yeah. I can't think of anything better.

Reply:
Commit pushed.